haskell-src/Concordium/Wasm.hs

```diff
@@ -290,6 +304,15 @@ moduleSourceLength = fromIntegral . BS.length . moduleSource
 newtype WasmModuleV (v :: WasmVersion) = WasmModuleV {wmvSource :: ModuleSource v}
     deriving (Eq, Show)
 
+-- Implement `FromJSON` and `ToJSON` instances for `WasmModuleV`.
+$( deriveJSON
+    defaultOptions
+        { AE.constructorTagModifier = firstLower,
+          AE.fieldLabelModifier = firstLower . dropWhile isLower
+        }
+    ''WasmModuleV
+ )
+
```

Comment:
I think giving the instance explicitly would be preferable, because it's short and it will be clearer. Also, I would suggest including the version in the serialization to be in line with the Serialize instance:

```suggestion
instance (IsWasmVersion v) => AE.ToJSON (WasmModuleV v) where
    toJSON (WasmModuleV ws) =
        AE.object
            [ "version" AE..= wasmVersionToWord (demoteWasmVersion (getWasmVersion @v)),
              "source" AE..= moduleSource ws
            ]

instance (IsWasmVersion v) => AE.FromJSON (WasmModuleV v) where
    parseJSON = AE.withObject "WasmModuleV" $ \obj -> do
        version <- obj AE..: "version"
        if wordToWasmVersion version == Just (demoteWasmVersion (getWasmVersion @v))
            then WasmModuleV . ModuleSource <$> obj AE..: "source"
            else fail $ "Expecting a " ++ show (getWasmVersion @v) ++ " module."

```